### PR TITLE
Add support for RPM repository indexing

### PR DESCRIPTION
**Pull request by cottsay**
This change should unblock trigger jobs and as long as the URL directory scheme matches debian (which is possible), should unblock status pages as well.

### CODE REVIEW

File: ros_buildfarm/common.py [hunk 1]
```diff
@@ -575,9 +576,20 @@ def get_packages_in_workspaces(workspace_roots, condition_context):
...
+        'rpm': get_rpm_repo_index,
+    }
+
     data = {}
     for target in targets:
-        index = get_debian_repo_index(
+        package_format = package_format_mapping.get(target.os_name)
+        get_index_method = get_index_methods.get(package_format)
+        if not get_index_method:
+            raise RuntimeError(
```
Comment (cottsay) [inline ros_buildfarm/common.py]: I feel like I'm going to end up with a lot of checks like this. Would you rather I just index directly into the maps and allow the `KeyError` to bubble up if something is missing?

Comment (dirk-thomas) [reply ros_buildfarm/common.py]: Raising a `KeyError` in this case (which should never happen) sounds fine to me.

Comment (cottsay) [reply ros_buildfarm/common.py]: OK, updated to remove the extra logic.

File: ros_buildfarm/status_page.py [hunk 2]
```diff
@@ -56,12 +56,15 @@ def build_release_status_page(
...
     for os_name in sorted(build_file.targets.keys()):
-        if os_name not in ['debian', 'ubuntu']:
+        if os_name not in ['debian', 'fedora', 'rhel', 'ubuntu']:
             continue
         for os_code_name in sorted(build_file.targets[os_name].keys()):
             targets.append(Target(os_name, os_code_name, 'source'))
             for arch in sorted(build_file.targets[os_name][os_code_name]):
                 targets.append(Target(os_name, os_code_name, arch))
+    if not targets:
+        print('The build file contains no supported targets')
```
Comment (dirk-thomas) [inline ros_buildfarm/status_page.py]: Should that (error) message go to `stderr`?

Comment (cottsay) [reply ros_buildfarm/status_page.py]: Done.

**Pull request merged**